internal/serialization/field_operations.go

```diff
@@ -0,0 +1,373 @@
+	return VARIABLE_SIZE
+}
+
+type ArrayOfTimestampFieldKindOperation struct{}
+
+func (ArrayOfTimestampFieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type TimestampWithTimezoneFieldKindOperation struct{}
+
+func (TimestampWithTimezoneFieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfTimestampWithTimezoneFieldKindOperation struct{}
+
+func (ArrayOfTimestampWithTimezoneFieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type CompactFieldKindOperation struct{}
+
+func (CompactFieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfCompactFieldKindOperation struct{}
+
+func (ArrayOfCompactFieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableBooleanFieldKindOperation struct{}
+
+func (NullableBooleanFieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableBooleanFieldKindOperation struct{}
+
+func (ArrayOfNullableBooleanFieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableInt8FieldKindOperation struct{}
+
+func (NullableInt8FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableInt8FieldKindOperation struct{}
+
+func (ArrayOfNullableInt8FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableInt16FieldKindOperation struct{}
+
+func (NullableInt16FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableInt16FieldKindOperation struct{}
+
+func (ArrayOfNullableInt16FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableInt32FieldKindOperation struct{}
+
+func (NullableInt32FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableInt32FieldKindOperation struct{}
+
+func (ArrayOfNullableInt32FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableInt64FieldKindOperation struct{}
+
+func (NullableInt64FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableInt64FieldKindOperation struct{}
+
+func (ArrayOfNullableInt64FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableFloat32FieldKindOperation struct{}
+
+func (NullableFloat32FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableFloat32FieldKindOperation struct{}
+
+func (ArrayOfNullableFloat32FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type NullableFloat64FieldKindOperation struct{}
+
+func (NullableFloat64FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+type ArrayOfNullableFloat64FieldKindOperation struct{}
+
+func (ArrayOfNullableFloat64FieldKindOperation) KindSizeInBytes() int32 {
+	return VARIABLE_SIZE
+}
+
+func FieldOperations(fieldKind pserialization.FieldKind) FieldKindOperation {
```

Comment:
I couldn't grasp why we need `FieldKindOperation` type. It seem to act as a mapping between `FieldKind` and the corresponding "kind size in bytes".
In this method we already map each type to something. Why not return the actual "kind size in bytes" directly as in
```
case pserialization.FieldKindBoolean:
    return 0
```
Is there another reason for `FieldKindOperation`'s existence?

Reply:
We will need to add more `FieldKindOperation`s after adding generic records. So, it is like prepration for generic records. Should not we do that?

Reply:
Oh I see, thanks! This implementation makes sense